deno/lib/helpers/parseUtil.ts

```diff
@@ -103,6 +103,7 @@ export const issueHelpers = (error: ZodError, params: ParseParams) => {
 
 export type ParseParams = {
   data: any;
+  userContext: any;
```

Comment:
I've only stumbled across this PR and Issue coming from looking into an alternative to Yup as I'm not too impressed with the loose typing, but would it be worth using a generic type here over `any`?
```ts
export type ParseParams<T extends {}> = {
  userContext: T
 // ...
```
And also in `RefinementCtx`/`superRefine`?
That way you could do something along the lines of
```ts
.superRefine<UserContext>((arg, ctx) => {
  ctx.userContext.inc();
  return arg;
});
```
Which would relieve the need to cast `ctx.userContext as UserContext`